xen: free domain's vcpu array

This was overlooked in fb442e2171 ("x86_64: allow more vCPU-s per
guest").

This is XSA-149.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Reviewed-by: Ian Campbell <ian.campbell@citrix.com>

Index: xen-4.1.6.1/xen/common/domain.c
===================================================================
--- xen-4.1.6.1.orig/xen/common/domain.c	2015-11-03 16:28:59.188546261 -0600
+++ xen-4.1.6.1/xen/common/domain.c	2015-11-03 16:29:27.756546183 -0600
@@ -670,6 +670,7 @@ static void complete_domain_destroy(stru
     xfree(d->pirq_to_evtchn);
 
     xsm_free_security_domain(d);
+    xfree(d->vcpu);
     free_domain_struct(d);
 
     send_guest_global_virq(dom0, VIRQ_DOM_EXC);
